src/main/java/com/google/step/datamanager/DealManager.java

```diff
@@ -10,11 +11,12 @@ public Deal createDeal(
       String end,
       String source,
       long posterId,
-      long restaurantId);
+      long restaurantId,
+      List<String> tagNames);
```

Comment:
It's a bit odd that we have two identifiers for tags, IDs and the names. Is this just because Datastore generates a numerical ID? If so, this is fine. Otherwise, we need to reconsider what the primary key for Tags are and maybe pass in Tag IDs here instead of Tag names.

Reply:
The names should be unique for each tag, so both the names and ID should be valid identifiers.
I passed in names here because the frontend allows the user to enter any string as a tag, so the servlet would receive the tagNames without the IDs.